### DISCUSSION TITLE
Hierarchical Risk Parity (HRP)

### DISCUSSION THREAD

**Discussion by Neoclassic28** [dcajasn/Riskfolio-Lib#33] [General]
Buen día Dany,
La librería que has desarrollado me parece muy buena e interesante, estuve haciendo uso del modelo HRP para un portafolio de 10 divisas aproximadamente, y me quedan las siguientes dudas:

1. Entiendo que el modelo ya agrupa por medio de Clusters a las divisas(se aprecia en los dendogramas), por lo que no es necesario que las divisas o activos que entren al portafolio no estén altamente correlacionado ¿es correcto?

2. Si mediante el modelo HRP obtengo los pesos óptimos solo con data de los retornos de los últimos 15 días, ¿significa que esos pesos los puedo mantener los siguientes 15 días?
Es decir, este Algoritmo desarrollado por López de Prado ¿tiene algún poder predictivo? o en todo caso cada cuanto se recomienda hacer un rebalanceo.

Nuevamente aprovecho para os felicitar por la excelente librería y quedo atento a tu cordial respuesta.

¡Saludos!

**Comment by dcajasn** [comment]
Hola @Neoclassic28 

Gracias por tu apoyo. Acerca de tus preguntas:

1. HRP no agrupa por clusters, solo crea el dendograma, y comienza dividir los pesos usando paridad de riesgo ingenua (pesos proporcionales al inverso del riesgo) de arriba hacia abajo partiendo el dendograma en partes iguales. Si quieres agrupar por clusters aprovechando la estructura del dendrogama debes usar el modelo HERC.
2. El algoritmo HRP hasta donde sé tiene como objetivo reducir el costo computacional de los modelos de optimización clásicos e incrementar la diversificación de los portafolios para mejorar su estabilidad. Respecto al rebalanceo, eso queda a criterio tuyo, ya que normalmente estos algoritmos asumen q el futuro se comportará como el periodo que utilizaste para calcular los pesos. Por otra parte, este tipo de algoritmos están pensados para rebalanceo mensual, trimestral, etc usando data de varios años como input; no estoy seguro como será su comportamiento en periodos tan cortos con data de alta frecuencia.

Saludos,
Dany

**Reply by Neoclassic28** [reply]
Buen día @dcajasn,

Gracias por vuestra pronta respuesta y aclaración.
Me parece muy interesante el modelo de agrupamiento por clusters "HERC".

Me gustaría comentarte que estuve haciendo algunas pruebas con ambos modelos y surgió la interrogante de si vuestra librería os permitiría obtener pesos óptimos con posiciones cortas( pesos negativos).

Y nuevamente, gracias de antemano.

iSaludos!

**Reply by dcajasn** [reply]
En los modelos de hierarchical clustering (HRP y HERC) por definicion no se permiten pesos negativos. Sin embargo, hay algunas variaciones a esos modelos que permiten  pesos negativos pero aun no las he implementado en Riskfolio-Lib. Estoy avanzando por esa parte pero creo que lo tendré en el mediano plazo.

**Reply by Neoclassic28** [reply]
Estimado @dcajasn,

Gracias por sus oportunas respuestas a mis interrogantes, sin duda aportan muchísimo a la comunidad.

Estuve revisando los ejemplos del modelo y en la sección de sensibilidad con las métricas de riesgos, me parecieron excelentes!
¿Tendrá alguna bibliografía donde pueda ahondar más acerca de estás métricas?
Especialmente la métrica ADD(Average Drawdown for compounded and uncompounded cumulative returns.)  de la cual  no encuentro mucha información en la red.

Gracias de antemano.
iSaludos!
